### PR TITLE
Fixed generated AutoMockable compilation issue due to generated variable names containing & character.  Added support for existential any for throwable errors.

### PR DESCRIPTION
**Pull request by pocketal**
1. fixed generated AutoMockable compilation issue due to generated variable names containing & character.
2. added support for existential any for throwable errors.

### CODE REVIEW

File: Templates/Tests/Expected/AutoMockable.expected [hunk 1]
```diff
@@ -1484,7 +1484,7 @@ class StaticMethodProtocolMock: StaticMethodProtocol {
 
     static func staticFunction(_ : String) -> String {
         staticFunctionStringStringCallsCount += 1
-        staticFunctionStringStringReceived =
```
Comment (art-divin) [inline Templates/Tests/Expected/AutoMockable.expected:1487]: This change is redundant, please revert

File: Templates/Tests/Generated/AutoMockable.generated.swift [hunk 2]
```diff
@@ -1090,7 +1090,7 @@ public class ProtocolWithMethodWithInoutParameterMock: ProtocolWithMethodWithIno
         executeParamInoutStringVoidCallsCount += 1
         executeParamInoutStringVoidReceivedParam = param
         executeParamInoutStringVoidReceivedInvocations.append(param)
-        executeParamInoutStringVoidClosure?(&param)
```
Comment (art-divin) [inline Templates/Tests/Generated/AutoMockable.generated.swift:1093]: Is this change intended?

**Comment by art-divin** [comment]
Thank you for finding the issue and creating the PR.

Don't forget to update Linux-specific files as well 👍🏻 - details in https://github.com/krzysztofzablocki/Sourcery/pull/1253

File: Templates/Tests/Generated/AutoMockable.generated.swift (hunk 2 above)
Comment (pocketal) [reply Templates/Tests/Generated/AutoMockable.generated.swift]: It's fixed

File: Templates/Tests/Expected/AutoMockable.expected (hunk 1 above)
Comment (pocketal) [reply Templates/Tests/Expected/AutoMockable.expected]: Done

**Comment by art-divin** [comment]
There are test failures, both on macOS and Linux, on macOS - `AutoMockable_template__generates_expected_code` - most likely TemplateSpec of some sort has a test which uses files you've modified (cannot be more specific at the moment).
On linux - `QuickSpec.AutoMockable template, generates expected code`. Seems to be the same test which fails.

For information, you could run unit tests locally in Xcode on macOS to see the failure and debug, if needed.

That being said, I see the change:
```
internal extension AutoCasesEnum {
  static let count: Int = 4
  static let allCases: [AutoCasesEnum] = [
    .north,
    .south,
    .east,
    .west
  ]
}
public extension AutoCasesHasAssociatedValuesEnum {
  static let count: Int = 2
}
internal extension AutoCasesOneValueEnum {
  static let count: Int = 1
  static let allCases: [AutoCasesOneValueEnum] = [
    .one
  ]
}
```

which  does not make much sense to be frank. Could you please double check and maybe revert this part?
Aside that, again, thank you for your contributions 👍🏻

**Comment by pocketal** [comment]
It's strange indeed I have the save test result with or without the changes.
I revert
